src/SprykerEco/Service/Computop/Model/Mapper/Computop.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/**
+ * MIT License
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Service\Computop\Model\Mapper;
+
+use Generated\Shared\Transfer\ComputopResponseHeaderTransfer;
+use Spryker\Shared\Kernel\Transfer\TransferInterface;
+use SprykerEco\Service\Computop\Model\AbstractComputop;
+use SprykerEco\Shared\Computop\ComputopConstants;
+
+class Computop extends AbstractComputop implements ComputopInterface
+{
+
+    const ITEMS_SEPARATOR = '|';
+    const ATTRIBUTES_SEPARATOR = '-';
+
+    /**
+     * @param \Spryker\Shared\Kernel\Transfer\TransferInterface $cardPaymentTransfer
+     *
+     * @return string
+     */
+    public function getMacEncryptedValue(TransferInterface $cardPaymentTransfer)
+    {
+        $macDataArray = [
+            $cardPaymentTransfer->getPayId(),
```

Comment:
Inside model logic that requires certain transfer fields to be present and non empty we recommend using ->requireX() syntax before calling the field.
That allows more explicit and early exceptions instead of silently doing too much until it evetually fails with unrelated error messages.